contrib/CONTRIBUTING.md

```diff
@@ -0,0 +1,232 @@
+# Contributing to Bittensor
+
+The following is a set of guidelines for contributing to Bittensor, which are hosted in the [Opentensor Organization](https://github.com/opentensor) on GitHub. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+## Table Of Contents
+
+1. [I don't want to read this whole thing, I just have a question!!!](#i-dont-want-to-read-this-whole-thing-i-just-have-a-question)
```

Comment:
The numbering here is bogus, even if markdown automatically fixes this for you. Same for other numbered lists.

Reply:
This markdown syntactic sugar is not as simple as you may think. While you are correct that the numbering will be prefilled _in general_, there are catches that are not expected.

When you end up mixing lists or styles, they won't render as expected:

````md
1. First element
1. Second element
```
Some code
```
1. Third element
1. Fourth element
````

renders as

1. First element
1. Second element
```
Some code
```
1. Third element
1. Fourth element

---

This is fixable ! You can actually ask the list to continue from your last point:

````md
1. First element
1. Second element
```
Some code
```
3. Third element
1. Fourth element
````

renders as

1. First element
1. Second element
```
Some code
```
3. Third element
1. Fourth element

---

But at this point you actually need to do the counting manually (keep the 3) in order to have the lists in order, thus having to do some manual numbering anyway. At this point the manual numbering is error prone since it's quite easy to add an element to the list mid way but not update the continuation:

````md
1. First element
1. New element
1. Second element
```
Some code
```
3. Third element
1. Fourth element
````

renders as

1. First element
1. New element
1. Second element
```
Some code
```
3. Third element
1. Fourth element

---

In my opinion there are good reasons to do manual counting for these lists, given that otherwise you will sooner or later have to mix styles, and the mixes are likely to be error prone. I am open to other opinions, but contrary to what your comment implies, I did not write this originally 'just because'.

Reply:
Thank you Adrian. This is informative 👍 

Can get behind manual counting philosophically based on the reasoning that most issue cases can be avoided this way.